### PR TITLE
Feature/community base location

### PR DESCRIPTION
**Pull request by Zoeshwa**

### CODE REVIEW

**Review by erezdrutin — APPROVED** [review]

File: apps/ui/src/apis/communities/fetchAllCommunities.ts [hunk 1]
```diff
@@ -1,60 +1,38 @@
...
+  const data = await graphqlRequest<AllCommunitiesData>(query);
+
+  const allCommunities = data.allCommunities.nodes.map(
+    async (node): Promise<Community> => {
+      const location =
+        node.lat && node.long ? await locationExtraction(node) : undefined;
+
+      const picturesUrl = node.userCommunitiesByCommunityId.nodes
+        .map((userCommunity) => userCommunity.userByUserId.profileImage)
+        .filter((url): url is string => url != null);
```
Comment (Kovler10) [inline apps/ui/src/apis/communities/fetchAllCommunities.ts:19]: (url: string) => url !== null) or !isNil(url)

File: apps/ui/src/apis/location/location.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const response = await axiosClient.get<LocationResult[]>(
+      '/api/v1/external/reverse-geocode',
+      {
+        params: { lat: coords.lat, lon: coords.lon },
+      },
+    );
+
+    if (response.data) {
+      const data = response.data;
+      if (!isArray(data)) {
```
Comment (Kovler10) [inline apps/ui/src/apis/location/location.ts:23]: let {data} = response;
if(!isArray(data)) data = [data];

Comment (Kovler10) [reply apps/ui/src/apis/location/location.ts:23]: and then you remove the check for is Array in the second if

**Review by Kovler10 — CHANGES_REQUESTED** [review]

File: apps/ui/src/apis/communities/fetchAllCommunities.ts (hunk 1 above)
Comment (Zoeshwa) [reply apps/ui/src/apis/communities/fetchAllCommunities.ts:19]: It doesn't work any other way

**Pull request merged**